challenges/data.js

```diff
@@ -0,0 +1,84 @@
+//backend challenge
+const express = require('express');
+const router = express.Router();
+var num = null;
+var arr = Array(5000);
+
+router.post('/data', (req, res, next)=>{
+    num =  req.body.numbers;
+    var i = 0, count=0;
+    var regulator = new RegExp('^\\d+$');
+    var fact = true;
+    try{
+        for(i in num){

```

Comment:
Hey John! Thanks for the submission. This answers the questions correctly, so kudos on that. I have a few things to provide feedback on, surrounding best practices/optimizations.

Reply:
Javascript has a few features in it that would allow you to answer the question of this block of code, which is "Does this array from the request contain exactly 500 numbers?" Is there a way that you could answer that question without explicitly looping through the values in the array?

Reply:
Hello! Thank you. Yes, we could also check the amount of data by checking the length of the array. To check all elements we have the array method called every(), within the function we could use a typeof comparison or the isNaN function. I hope I'm understanding the question correctly.